thrust/device_make_unique.h

```diff
@@ -39,16 +39,18 @@ THRUST_NAMESPACE_BEGIN
 template <typename T, typename... Args>
 __host__
 auto device_make_unique(Args&&... args)
-  -> decltype(
+  THRUST_TRAILING_RETURN(decltype(
     uninitialized_allocate_unique<T>(device_allocator<T>{})
-  )
+  ))
 {
-  // FIXME: This is crude - we construct an unnecessary T on the host for 
+#if !defined(THRUST_DOXYGEN) // This causes Doxygen to choke for some reason.
```

Comment:
Confirm whether this is still needed with `THRUST_TRAILING_RETURN`.

Reply:
This is still needed, otherwise Doxygen incorrectly reads a function declaration in the body of this function.